Can't cross-compile rust lib from linux to windows

## 🐛 Bug Reports

When reporting a bug, please provide the following information. If this is not a bug report you can just discard this template.

### 🌍 Environment

 - Your operating system and version: Debian Testing
 - Your python version: 3.7.4
 - How did you install python (e.g. apt or pyenv)? Did you use a virtualenv?: apt
 - Your rust version (`rustc --version`): 1.40.0-nightly (95f437b3c 2019-10-27)
 - Are you using the latest pyo3 version? Have you tried using latest master (replace `version = "0.x.y"` with `git = "https://github.com/PyO3/pyo3")?` yes

### 💥 Reproducing

I'm trying to compile [this lib](https://github.com/DrInfy/sc2-pathlib) on linux for windows.

Running with `cargo build --release --target "x86_64-pc-windows-gnu" -v`, I'm getting this error:
```
error: linking with `x86_64-w64-mingw32-gcc` failed: exit code: 1
  |
  = note: "x86_64-w64-mingw32-gcc" "-Wl,--enable-long-section-names" "-fno-use-linker-plugin" "-Wl,--nxcompat" "-nostdlib" "-m64" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/dllcrt2.o" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/rsbegin.o" "-L" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.0.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.1.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.10.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.11.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.12.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.13.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.14.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.15.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.2.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.3.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.4.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.5.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.6.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.7.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.8.rcgu.o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.sc2pathlib.a45jfogg-cgu.9.rcgu.o" "-o" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.dll" "-Wl,--version-script=/tmp/rustcINaZNj/list" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.4qrj5y82yo4grzjd.rcgu.o" "-Wl,--gc-sections" "-nodefaultlibs" "-L" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps" "-L" "/home/armageddon/projects/sc2-pathlib/target/release/deps" "-L" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib" "-Wl,-Bstatic" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libpathfinding-de2d6bd9c2051d11.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libfixedbitset-57795ffa7b25cc7f.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libitertools-49e282049a7e60d6.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libeither-69f3815dfbd07322.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libindexmap-c7188f68cefb1405.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libpyo3-5011b13c2f54e62b.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libnum_traits-20dc20a50dc118ef.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libspin-9b6bb5af72046670.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libunindent-07f758d7f0b54cfb.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/liblibc-ba8355680f151b78.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libinventory-2adc9d42c32988f9.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libindoc-91148f62fc14448a.rlib" "/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libpaste-31f65bf4a5d1b11e.rlib" "-Wl,--start-group" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libstd-747648f71118c692.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libpanic_unwind-3cf77e8c834e1515.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libhashbrown-37849e1cdf2d5ce0.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/librustc_std_workspace_alloc-ea5d6bc0ec7058c9.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libbacktrace-b5ec048bcf1e3694.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libbacktrace_sys-f21dfcd83a2909fb.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/librustc_demangle-c2033e316e103224.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libunwind-fe6eb7460b9ef950.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libcfg_if-12ebe9a4ac4ed76a.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/liblibc-6967320188af24fd.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/liballoc-318fbdaec35b4968.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/librustc_std_workspace_core-07539409623004fc.rlib" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libcore-3a26d76fb9e35800.rlib" "-Wl,--end-group" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libcompiler_builtins-c948b6664b7af323.rlib" "-Wl,-Bdynamic" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-lpythonXY" "-ladvapi32" "-lws2_32" "-luserenv" "-Wl,-Bstatic" "-lgcc_eh" "-lpthread" "-shared" "-Wl,--out-implib,/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/sc2pathlib.dll.lib" "-Wl,-Bdynamic" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-lmsvcrt" "-luser32" "-lkernel32" "/home/armageddon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/rsend.o"
  = note: /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
          collect2: error: ld returned 1 exit status
          

error: aborting due to previous error

error: could not compile `sc2pathlib`.

Caused by:
  process didn't exit successfully: `rustc --edition=2018 --crate-name sc2pathlib src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type cdylib --emit=dep-info,link -C opt-level=3 -C metadata=c436664892c0a209 --out-dir /home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps --target x86_64-pc-windows-gnu -C ar=x86_64-w64-mingw32-gcc-ar -C linker=x86_64-w64-mingw32-gcc -L dependency=/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps -L dependency=/home/armageddon/projects/sc2-pathlib/target/release/deps --extern pathfinding=/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libpathfinding-de2d6bd9c2051d11.rlib --extern pyo3=/home/armageddon/projects/sc2-pathlib/target/x86_64-pc-windows-gnu/release/deps/libpyo3-5011b13c2f54e62b.rlib` (exit code: 1)
```

@davidhewitt Just tried it now, didn't work. however, before the output contained 
```
/usr/bin/x86_64-w64-mingw32-ld: cannot find -lpythonXY
```
but now it's 
```
/usr/bin/x86_64-w64-mingw32-ld: cannot find -lpython3.6m
```
`XY` became the python version.
Also notice i had to use a nightly build of rust.

This is my `Cargo.toml` in case i'm doing something wrong
```
[dependencies]
pyo3 = { git = "https://github.com/davidhewitt/pyo3", branch = "fix-cross-compile" }
```

--- a/build.rs
+++ b/build.rs
@@ -171,7 +171,6 @@ fn cross_compiling() -> Result<Option<CrossCompileConfig>> {
 ///
 /// (hrm, this is sort of re-implementing what distutils does, except
 /// by passing command line args instead of referring to a python.h)
-#[cfg(not(target_os = "windows"))]
 static SYSCONFIG_FLAGS: [&str; 7] = [
     "Py_USING_UNICODE",
     "Py_UNICODE_WIDE",
@@ -443,8 +442,11 @@ fn load_cross_compile_info(
 /// Examine python's compile flags to pass to cfg by launching
 /// the interpreter and printing variables of interest from
 /// sysconfig.get_config_vars.
-#[cfg(not(target_os = "windows"))]
 fn get_config_vars(python_path: &Path) -> Result<HashMap<String, String>> {
+    if env::var("CARGO_CFG_TARGET_OS").unwrap() == "windows" {
+        return get_config_vars_windows(python_path);
+    }
+
     let mut script = "import sysconfig; \
                       config = sysconfig.get_config_vars();"
         .to_owned();
@@ -478,8 +480,7 @@ fn get_config_vars(python_path: &Path) -> Result<HashMap<String, String>> {
     Ok(fix_config_map(all_vars))
 }
 
-#[cfg(target_os = "windows")]
-fn get_config_vars(_: &Path) -> Result<HashMap<String, String>> {
+fn get_config_vars_windows(_: &Path) -> Result<HashMap<String, String>> {
     // sysconfig is missing all the flags on windows, so we can't actually
     // query the interpreter directly for its build flags.
     //
@@ -567,9 +568,15 @@ fn get_library_link_name(version: &PythonVersion, ld_version: &str) -> String {
     }
 }
 
-#[cfg(not(target_os = "macos"))]
-#[cfg(not(target_os = "windows"))]
 fn get_rustc_link_lib(config: &InterpreterConfig) -> Result<String> {
+    match env::var("CARGO_CFG_TARGET_OS").unwrap().as_str() {
+        "windows" => get_rustc_link_lib_windows(config),
+        "macos" => get_rustc_link_lib_macos(config),
+        _ => get_rustc_link_lib_unix(config),
+    }
+}
+
+fn get_rustc_link_lib_unix(config: &InterpreterConfig) -> Result<String> {
     if config.shared {
         Ok(format!(
             "cargo:rustc-link-lib={}",
@@ -583,7 +590,6 @@ fn get_rustc_link_lib(config: &InterpreterConfig) -> Result<String> {
     }
 }
 
-#[cfg(target_os = "macos")]
 fn get_macos_linkmodel(config: &InterpreterConfig) -> Result<String> {
     // PyPy 3.6 ships with a shared library, but doesn't have Py_ENABLE_SHARED.
     if config.version.implementation == PythonInterpreterKind::PyPy {
@@ -604,8 +610,7 @@ else:
     Ok(out.trim_end().to_owned())
 }
 
-#[cfg(target_os = "macos")]
-fn get_rustc_link_lib(config: &InterpreterConfig) -> Result<String> {
+fn get_rustc_link_lib_macos(config: &InterpreterConfig) -> Result<String> {
     // os x can be linked to a framework or static or dynamic, and
     // Py_ENABLE_SHARED is wrong; framework means shared library
     let link_name = get_library_link_name(&config.version, &config.ld_version);
@@ -617,8 +622,7 @@ fn get_rustc_link_lib(config: &InterpreterConfig) -> Result<String> {
     }
 }
 
-#[cfg(target_os = "windows")]
-fn get_rustc_link_lib(config: &InterpreterConfig) -> Result<String> {
+fn get_rustc_link_lib_windows(config: &InterpreterConfig) -> Result<String> {
     // Py_ENABLE_SHARED doesn't seem to be present on windows.
     Ok(format!(
         "cargo:rustc-link-lib=pythonXY:{}",
@@ -733,7 +737,7 @@ fn configure(interpreter_config: &InterpreterConfig) -> Result<String> {
         println!("{}", get_rustc_link_lib(&interpreter_config)?);
         if let Some(libdir) = &interpreter_config.libdir {
             println!("cargo:rustc-link-search=native={}", libdir);
-        } else if cfg!(target_os = "windows") {
+        } else if target_os == "windows" {
             println!(
                 "cargo:rustc-link-search=native={}\\libs",
                 interpreter_config.base_prefix
@@ -818,9 +822,6 @@ fn main() -> Result<()> {
     // Detecting if cross-compiling by checking if the target triple is different from the host
     // rustc's triple.
     let (interpreter_config, mut config_map) = if let Some(paths) = cross_compiling()? {
-        // If cross compiling we need the path to the cross-compiled include dir and lib dir, else
-        // fail quickly and loudly
-
         load_cross_compile_info(paths)?
     } else {
         find_interpreter_and_get_config()?

